File: dotNet/LSP/RefactoredSolid/LSP/Example2/Domain/PayPalPayment.cs
using RefactoredSolid.LSP.Example2.External;

namespace RefactoredSolid.LSP.Example2.Domain
{
    public class PayPalPayment : Payment
    {
        private readonly string _username;
        private readonly string _password;

        public PayPalPayment(string username, string password)
        {
            _username = username;
            _password = password;
        }

        public override bool Refund(decimal amount, string transactionId)
        {
            PayPalWebService payPalWebService = new PayPalWebService();
            string token = payPalWebService.GetTransactionToken(_username, _password);
            string response = payPalWebService.MakeRefund(amount, transactionId, token);

            return response.Contains("Authorized");
        }
    }
}

File: java/liskov-plus-refactor/src/main/java/no/bouvet/solid/lsp/domain/PayPalPayment.java
package no.bouvet.solid.lsp.domain;

import java.math.BigDecimal;

import no.bouvet.solid.lsp.external.PayPalWebService;

public class PayPalPayment extends Payment {

    private String username;
    private String password;

    public PayPalPayment(String username, String password) {
        this.username = username;
        this.password = password;
    }

    @Override
    public Boolean refund(BigDecimal amount, String transactionId) {
        PayPalWebService payPalWebService = new PayPalWebService();
        String token = payPalWebService.getTransactionToken(username, password);
        String response = payPalWebService.makeRefund(amount, transactionId, token);
        return response.contains("Authorized");
    }

}
